tests/CMakeLists.txt

```diff
@@ -94,13 +94,65 @@ set(KEM_TESTS example_kem kat_kem test_kem test_kem_mem speed_kem vectors_kem)
 add_executable(example_sig example_sig.c)
 target_link_libraries(example_sig PRIVATE ${TEST_DEPS})
 
+set(FUZZ_SIG_ALGORITHMS 
+    dilithium_2
```

Comment:
This list seems to be manually curated, right? This means it has to be equally manually updated whenever an upstream change occurs -- this is not good given the breadth of algorithms supported. I'd therefore strongly advocate the use of the `copy_from_upstream` logic to automatically build this list to avoid future breakage and continued, error-prone manual work.

Reply:
I found the logic around copy_from_upstream difficult to follow. I've instead just used the fuzzing data to select the algorithm to use at runtime via [OQS_SIG_alg_identifier](https://github.com/open-quantum-safe/liboqs/pull/1955/files#diff-5256ad59f90fd8a1fdf808aecf506ae6c9349ec5e67d87f6259bc4610b1f11e4R44). This should acheive approximately the same thing and reduce the number of fuzz harness executables to one.

Reply:
Yes, iterating through `OQS_SIG_alg_identifier`  looks like an OK approach to me. But is it then still necessary to list all algs manually here? This would mandate manual changes down the road even though `OQS_SIG_alg_identifier` delivers the current list, right?

Reply:
As far as I understand there would be no manual effort required to update this when algorithms are added/removed. If I'm reading [the docstring for OQS_SIG_alg_identifier](https://github.com/open-quantum-safe/liboqs/blob/3c8bde1b08ba068ebbde738165bf6f8d2ba385ce/src/sig/sig.h#L146) correctly it takes an index (from 0 to OQS_SIG_alg_identifier) and returns the string-type algorithm name which I've then passed into `OQS_SIG_new` [here](https://github.com/open-quantum-safe/liboqs/pull/1955/files#diff-5256ad59f90fd8a1fdf808aecf506ae6c9349ec5e67d87f6259bc4610b1f11e4R46). Under the hood it looks like it's [leveraging the existing code-gen support](https://github.com/open-quantum-safe/liboqs/blob/3c8bde1b08ba068ebbde738165bf6f8d2ba385ce/src/sig/sig.c#L17). So there is no need to manually update anything. There is also [OQS_SIG_algs_length](https://github.com/open-quantum-safe/liboqs/blob/3c8bde1b08ba068ebbde738165bf6f8d2ba385ce/src/sig/sig.h#L134) that is already generated as well which specifies the range of acceptable inputs to `OQS_SIG_alg_identifier`. In short there seems to be enough information around that is already generated, that I can leverage rather than adding additional logic to the code-generation system.

Also just as a note, I'm not really iterating over `OQS_SIG_alg_identifier` rather I'm taking a portion of the fuzzed data inputs [writing that into an integer](https://github.com/open-quantum-safe/liboqs/pull/1955/files#diff-5256ad59f90fd8a1fdf808aecf506ae6c9349ec5e67d87f6259bc4610b1f11e4R35-R39) then [capping the max-value of the integer at OQS_SIG_algs_length](https://github.com/open-quantum-safe/liboqs/pull/1955/files#diff-5256ad59f90fd8a1fdf808aecf506ae6c9349ec5e67d87f6259bc4610b1f11e4R39). This effectively allows the fuzzer to "choose" which algorithm to fuzz. This ensures that there is a large amount of independant "fuzzed data" available when fuzzing each algorithm. I'm happy to elaborate further, as to why this is better than iterating over all algorithms if you are interested.

Reply:
Thanks for the explanation @nathaniel-brough . The list that started this discussion is now gone from the PR, so is my concern.